src/main/java/com/faforever/client/fa/RatingMode.java

```diff
@@ -1,7 +1,20 @@
 package com.faforever.client.fa;
 
+import com.faforever.client.game.KnownFeaturedMod;
+
 public enum RatingMode {
   GLOBAL,
   LADDER_1V1,
-  NONE
+  LADDER_2V2,
+  NONE;
+
+  public KnownFeaturedMod getCorrespondingFeatureMod() {
+    switch (this) {
+      case LADDER_1V1: return KnownFeaturedMod.LADDER_1V1;
+      case LADDER_2V2: return KnownFeaturedMod.LADDER_2V2;
+      default: throw new UnsupportedOperationException("FeatureMod for this rating mod is unsupported or not created yet");
+    }
+  }
```

Comment:
We should try to avoid hardcoding references to anything besides global and ladder1v1. There is also no one-to-one relationship between rating mode and game mode like this. We are trying to move away from having a special `ladder1v1` gamemode and having ladder games just be played on the `faf` gamemode, and have the rating information stored separately.

Reply:
Then how do u know that a featured mod is a auto match mod :D

Reply:
If we want to avoid it, we have to receive them from a server dynamically. To do that we have to create a new rest API endpoint which will provide the client with all available ladder queue types. Each queue type will encapsulate some information like name, description, required party size, and technical name for the game mode. This technical name will be used in composing a message which will be sent to a server when a player hits the search (play) button.

Reply:
Oh, I'm sorry, maybe I just didn't get it or missed smth? You mentioned "matchmaker_info" message. What this message is? Are queue types already implemented on a server side or rest api side?  I do not fully understand the client-server communication process yet (what are these messages are and how can I get a response after writing a message), but I'm working on it.

Reply:
@axel1200 there is mod such thing as an “automatch mod”. Automatch is independent of the featured mod and happens on a per game basis (the server sends the corresponding `InitMode.AUTO_MATCH` in the `game_launch` message. In theory it could even be a coop or nomads game, the client should be able to handle that.

Queues have been implemented but not merged/deployed yet. They are sitting on my team matchmaking branch. I will try to merge it soon.

Also be aware of the distinction between API and server. The server has a connection to the client at all times and can send events at any time. But we are trying to keep more generic data queries which don’t depend on what other players are doing to the API.

Reply:
> Also be aware of the distinction between API and server. The server has a connection to the client at all times and can send events at any time. But we are trying to keep more generic data queries which don’t depend on what other players are doing to the API.

Yea I've already realized that. It was not very clear from the beginning. It would be great to have some sort of basic documentation for newcomers with some real code examples.

Reply:
Please leave complete information about queue type content here when it's deployed.